src/panel/PoiPanel.jsx

```diff
@@ -0,0 +1,345 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Telemetry from '../libs/telemetry';
+import nconf from '@qwant/nconf-getter';
+import layouts from './layouts.js';
+
+import PoiHeader from 'src/panel/poi/PoiHeader';
+import PoiTitleImage from 'src/panel/poi/PoiTitleImage';
+import OpeningHour from 'src/components/OpeningHour';
+import OsmContribution from 'src/components/OsmContribution';
+import PoiBlockContainer from './poi_bloc/PoiBlockContainer';
+
+import Store from '../adapters/store';
+import MasqFavoriteModal from '../modals/masq_favorite_modal';
+import CategoryService from '../adapters/category_service';
+import { openShareModal } from 'src/modals/ShareModal';
+
+const store = new Store();
+const masqFavoriteModal = new MasqFavoriteModal();
+
+async function isPoiFavorite(poi) {
+  try {
+    const storePoi = await store.has(poi);
+    if (storePoi) {
+      return true;
+    }
+  } catch (e) {
+    return false;
+  }
+  return false;
+}
+
+export default class PoiPanel extends React.Component {
+  static propTypes = {
+    poi: PropTypes.object,
+    isFromFavorite: PropTypes.bool,
+    isFromCategory: PropTypes.bool,
+    sourceCategory: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    console.log(props);
+    this.state = { isCollapsed: true };
+    this.categories = CategoryService.getCategories();
+    this.isDirectionActive = nconf.get().direction.enabled;
+    this.isMasqEnabled = nconf.get().masq.enabled;
+
+    this.card = true;
+    isPoiFavorite(this.props.poi).then(x => this.props.poi.stored = x);
```

Comment:
React components shouldn't have side effects on data, meaning you must not modify what is passed as props. This should probably be done outside, or as state.